### PR TITLE
Typos

### PR DESCRIPTION
**Pull request by kevaundray**
In some places, s was being used and in others z. This PR changes the spelling to be consistent and corrects a few typos. 

If the spelling change is not needed, I can cherry-pick these commits out.

### CODE REVIEW

**Comment by steven-ecc** [comment]
Hi! Thanks for this contribution.

Could you please sign the [contributor agreement](https://github.com/zcash/halo2/blob/main/Contributor_Agreement) and send it to me at steven@electriccoin.co? Signing the agreement allows us to redistribute the code later under another license if we need to, without having to get your permission. This is similar to how contributor agreements work in Apache software projects as well.

File: book/src/background/fields.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 A fundamental component of many cryptographic protocols is the algebraic structure known
 as a [field]. Fields are sets of objects (usually numbers) with two associated binary
 operators $+$ and $\times$ such that various [field axioms][field-axioms] hold. The real
-numbers $\mathbb{R}$ are an example of a field with an uncountably infinite number of
+numbers $\mathbb{R}$ are an example of a field with an uncountable number of
```
Comment (therealyingtong) [inline book/src/background/fields.md:6]:
```suggestion
numbers $\mathbb{R}$ are an example of a field with an uncountably infinite number of
```
We do mean to say "uncountably infinite" here: https://mathworld.wolfram.com/UncountablyInfinite.html

**Review by therealyingtong — CHANGES_REQUESTED** [review]

File: book/src/background/fields.md (hunk 1 above)
Comment (kevaundray) [reply book/src/background/fields.md:6]: Yep, this was just a small nit for readability, as uncountability implies the set is infinite. 

I can change it back if you think the previous reading is clearer or better terminology wise.

File: book/src/background/fields.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
...
 elements.
```
Comment (daira) [inline book/src/background/fields.md:7]:
```suggestion
numbers $\mathbb{R}$ are an example of a field with uncountably many elements.
```

File: book/src/SUMMARY.md [hunk 3]
```diff
@@ -3,7 +3,7 @@
 [halo2](README.md)
 - [Concepts](concepts.md)
   - [Proof systems](concepts/proofs.md)
-  - [UltraPLONK Arithmetization](concepts/arithmetization.md)
+  - [UltraPLONK arithmetization](concepts/arithmetization.md)
```
Comment (daira) [inline book/src/SUMMARY.md:6]: This is intentionally capitalized because it has the abbreviation UPA.

File: book/src/SUMMARY.md [hunk 4]
```diff
@@ -31,6 +31,6 @@
   - [Polynomials](background/polynomials.md)
   - [Cryptographic groups](background/groups.md)
   - [Elliptic curves](background/curves.md)
-  - [UltraPLONK arithmetisation](background/upa.md)
+  - [UltraPLONK arithmetization](background/upa.md)
```
Comment (daira) [inline book/src/SUMMARY.md:34]:
```suggestion
  - [UltraPLONK Arithmetization](background/upa.md)
```

**Comment by daira** [comment]
Merged into #320. I don't think we need a contributor agreement for fixing typos.

**Pull request closed without merging**